otelcol/command_print.go

```diff
@@ -4,55 +4,205 @@
-	"otelcol.printInitialConfig",
+	featureGateName,
 	featuregate.StageAlpha,
 	featuregate.WithRegisterFromVersion("v0.120.0"),
-	featuregate.WithRegisterDescription("if set to true, turns on the print-initial-config command"),
+	featuregate.WithRegisterDescription("if set to true, enable the print-config command"),
 )
 
-// newConfigPrintSubCommand constructs a new config print sub command using the given CollectorSettings.
-func newConfigPrintSubCommand(set CollectorSettings, flagSet *flag.FlagSet) *cobra.Command {
+// newConfigPrintSubCommand constructs a new print-config command using the given CollectorSettings.
+func newConfigPrintSubCommand(set CollectorSettings, flagSet *flag.FlagSet, stdout, stderr io.Writer) *cobra.Command {
+	var outputFormat string
+	var mode string
+	var validate bool
+
 	cmd := &cobra.Command{
-		Use:   "print-initial-config",
-		Short: "Prints the Collector's configuration in YAML format after all config sources are resolved and merged",
-		Long:  `Note: This command prints the final yaml configuration before it is unmarshaled into config structs, which may contain sensitive values.`,
-		Args:  cobra.ExactArgs(0),
+		Use:     "print-config",
+		Aliases: []string{"print-initial-config"},
+		Short:   "Prints the Collector's configuration in the specified mode",
+		Long: `Prints the Collector's configuration with different levels of processing:
+
+- redacted: Shows the resolved configuration with sensitive data redacted (default)
+- unredacted: Shows the resolved configuration with all sensitive data visible
+
+The output prints in YAML by default. To print JSON use --format=json,
+however this is considered unstable.
+
+Validation is enabled by default, as a safety measure.
+
+All modes are experimental requiring otelcol.printInitialConfig feature gate.`,
+		Args: cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, _ []string) error {
-			if !printCommandFeatureFlag.IsEnabled() {
-				return errors.New("print-initial-config is currently experimental, use the otelcol.printInitialConfig feature gate to enable this command")
-			}
-			err := updateSettingsUsingFlags(&set, flagSet)
-			if err != nil {
-				return err
+			pc := printContext{
+				cmd:          cmd,
+				stdout:       stdout,
+				stderr:       stderr,
+				set:          set,
+				outputFormat: outputFormat,
+				validate:     validate,
 			}
-			resolver, err := confmap.NewResolver(set.ConfigProviderSettings.ResolverSettings)
-			if err != nil {
-				return fmt.Errorf("failed to create new resolver: %w", err)
-			}
-			conf, err := resolver.Resolve(cmd.Context())
-			if err != nil {
-				return fmt.Errorf("error while resolving config: %w", err)
-			}
-			b, err := yaml.Marshal(conf.ToStringMap())
-			if err != nil {
-				return fmt.Errorf("error while marshaling to YAML: %w", err)
-			}
-			fmt.Printf("%s\n", b)
-			return nil
+			return pc.configPrintSubCommand(flagSet, mode)
 		},
 	}
+	cmd.SetOut(stdout)
+	cmd.SetErr(stderr)
```

Comment:
If this is only for testing (otherwise I don't see a reason to have them) can you configure them after cmd is returned from this func instead of accepting as params to the func?